src/message/__tests__/fetchActions-test.js

```diff
@@ -21,6 +22,55 @@ describe('fetchActions', () => {
     fetch.reset();
   });
 
+  describe('isFetchNeededAtAnchor', () => {
+    test("false if we're caught up, even if there are no messages", () => {
+      const state = eg.reduxState({
```

Comment:
Let's squash this and the next commit together. That makes it a bit easier to read this commit, because it becomes visible that a lot of the code is simply moved, or moved with small edits.